src/compiler/scala/tools/nsc/ast/DocComments.scala

```diff
@@ -311,7 +313,8 @@ trait DocComments { self: Global =>
 
       searchList collectFirst { case x if defs(x) contains vble => defs(x)(vble) } match {
         case Some(str) if str startsWith "$" => lookupVariable(str.tail, site)
-        case res                             => res orElse lookupVariable(vble, site.owner)
+        case s @ Some(str)                   => s
+        case None                            => lookupVariable(vble, site.owner)
       }
   }
```

Comment:
hand inlined `orElse` to enable tail call eliminiation

Reply:
`case s @ Some(_)` avoids unused var. Or `case s @ Some(str @ _)` for baroque effect.

Recursion didn't come up in the discussions about "avoid match on options".